### PR TITLE
fix: auth requests use org and user names if present

### LINKED ISSUE #22261: /api/v2/authorizations/?user query parameter does not filter authorizations
__Steps to reproduce:__
1. Add a user, .e.g. username = "myusername".
2. Add authorizations for the user.
3. Repeat for additional users.
4. Send a GET request with `?user=myusername` to the http://localhost:8086/api/v2/authorizations/ [endpoint](https://docs.influxdata.com/influxdb/v2.0/api/#operation/GetAuthorizations). 

__Expected behavior:__
According to the [API docs](https://docs.influxdata.com/influxdb/v2.0/api/#operation/GetAuthorizations) and [authorization/http_client.go](https://github.com/influxdata/influxdb/blob/master/authorization/http_client.go#L43),  queryparameter `user` should filter authorizations to only myusername's auths.

__Actual behavior:__
The API returns all authorizations without filtering. See attached curl request and response.
[api-v2-user-auths.log](https://github.com/influxdata/influxdb/files/7015800/api-v2-user-auths.log)

__Environment info:__

* System info: Run `uname -srm` and copy the output here

> Darwin 20.6.0 x86_64

* InfluxDB version: Run `influxd version` and copy the output here

> InfluxDB 2.0.8 (git: e91d41810f) build_date: 2021-08-13T17:55:41Z

### PR DESCRIPTION
**Pull request by wbaker85**
Closes #22261

Fixes requests to `/api/v2/authorizations` so that the query parameters of `user` and/or `org` will be used to populate the filter with the respective ID's if `user`/`org` are present but `userID`/`orgID` are not in the query.

Giving precedence to the `ID` parameters if they are present over the "name" parameters seems to be consistent with how other resource endpoints work. For example, if you provide both a `user` and a `userID`, the `userID` will be used to filter the authorizations.

### CODE REVIEW

**Review by gwossum — COMMENTED** [review]
Looks like it should work as described. Are there test cases that test if an invalid user or org are in the query? Couldn't tell if there were any tests cases that did that.

File: CHANGELOG.md [hunk 1]
```diff
@@ -83,6 +83,7 @@ This release adds an embedded SQLite database for storing metadata required by t
 1. [#22228](https://github.com/influxdata/influxdb/pull/22228): influxdb2 packages should depend on curl
 1. [#22211](https://github.com/influxdata/influxdb/pull/22211): Prevent scheduling an inactivated tasks after updating it
 1. [#22235](https://github.com/influxdata/influxdb/pull/22235): Avoid compaction queue stats flutter
+1. [#22272](https://github.com/influxdata/influxdb/pull/22272): Requests to `/api/v2/authorizations` fillter correctly on `org` and `user` parameters
```
Comment (MarcusMcKibben) [inline CHANGELOG.md:86]: There's a typo in the world "fillter".  Simple change.

Comment (wbaker85) [reply CHANGELOG.md]: Fixed, thanks for the heads-up!

**Comment by wbaker85** [comment]
> Looks like it should work as described. Are there test cases that test if an invalid user or org are in the query? Couldn't tell if there were any tests cases that did that.

It goes through the tenant service, which will cause the handler to return an error (404 HTTP response) if the request user or org name doesn't exist. There aren't tests directly here for that, but the tenant service does have tests for [org not found](https://github.com/influxdata/influxdb/blob/467040a072511911bee79af9c861feaa6e0377be/tenant/storage_org_test.go#L120-L122) and [user not found](https://github.com/influxdata/influxdb/blob/467040a072511911bee79af9c861feaa6e0377be/tenant/storage_user_test.go#L93-L95).

**Review by gwossum — APPROVED** [review]
I've seen enough. Approved.

**Pull request merged**